pkg/solana/txm/txm_integration_test.go

```diff
@@ -0,0 +1,186 @@
+		name                    string
+		txExpirationRebroadcast bool
+		useValidBlockHash       bool
+		expectRebroadcast       bool
+		expectTransactionStatus types.TransactionStatus
+	}
+
+	testCases := []TestCase{
+		{
+			name:                    "WithRebroadcast",
+			txExpirationRebroadcast: true,
+			useValidBlockHash:       false,
+			expectRebroadcast:       true,
+			expectTransactionStatus: types.Finalized,
+		},
+		{
+			name:                    "WithoutRebroadcast",
+			txExpirationRebroadcast: false,
+			useValidBlockHash:       false,
+			expectRebroadcast:       false,
+			expectTransactionStatus: types.Failed,
+		},
+		{
+			name:                    "ConfirmedBeforeRebroadcast",
+			txExpirationRebroadcast: true,
+			useValidBlockHash:       true,
+			expectRebroadcast:       false,
+			expectTransactionStatus: types.Finalized,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			ctx, client, txmInstance, senderPubKey, receiverPubKey, observer := setup(t, url, tc.txExpirationRebroadcast)
+
+			// Record initial balance
+			initSenderBalance, err := client.Balance(ctx, senderPubKey)
+			require.NoError(t, err)
+			const amount = 1 * solana.LAMPORTS_PER_SOL
+
+			// Create and enqueue tx
+			txID := tc.name
+			tx, lastValidBlockHeight := createTransaction(ctx, t, client, senderPubKey, receiverPubKey, amount, tc.useValidBlockHash)
+			require.NoError(t, txmInstance.Enqueue(ctx, "", tx, &txID, lastValidBlockHeight))
+
+			// Wait for the transaction to reach the expected status
+			require.Eventually(t, func() bool {
+				status, statusErr := txmInstance.GetTransactionStatus(ctx, txID)
+				if statusErr != nil {
+					return false
+				}
+				return status == tc.expectTransactionStatus
+			}, 60*time.Second, 1*time.Second, "Transaction should eventually reach expected status")
+
+			// Verify balances
+			finalSenderBalance, err := client.Balance(ctx, senderPubKey)
+			require.NoError(t, err)
+			finalReceiverBalance, err := client.Balance(ctx, receiverPubKey)
+			require.NoError(t, err)
+
+			if tc.expectTransactionStatus == types.Finalized {
+				require.Less(t, finalSenderBalance, initSenderBalance, "Sender balance should decrease")
+				require.Equal(t, amount, finalReceiverBalance, "Receiver should receive the transferred amount")
+			} else {
+				require.Equal(t, initSenderBalance, finalSenderBalance, "Sender balance should remain the same")
+				require.Equal(t, uint64(0), finalReceiverBalance, "Receiver should not receive any funds")
+			}
+
+			// Verify rebroadcast logs
+			rebroadcastLogs := observer.FilterMessageSnippet("rebroadcast transaction sent").All()
+			rebroadcastLogs2 := observer.FilterMessageSnippet("transaction expired, rebroadcasting").All()
+			if tc.expectRebroadcast {
+				require.NotEmpty(t, rebroadcastLogs, "Expected rebroadcast log message not found")
+				require.NotEmpty(t, rebroadcastLogs2, "Expected rebroadcast log message not found")
```

Comment:
I think it might be worth also ensuring that we only rebroadcast once here. Could we maybe leverage `Len()` and check that it's explicitly 1 here?
```suggestion
			rebroadcastLogs := observer.FilterMessageSnippet("rebroadcast transaction sent").Len()
			rebroadcastLogs2 := observer.FilterMessageSnippet("transaction expired, rebroadcasting").Len()
			if tc.expectRebroadcast {
				require.Equal(t, 1, rebroadcastLogs, "Expected rebroadcast log message not found")
				require.Equal(t, 1, rebroadcastLogs2, "Expected rebroadcast log message not found")
```

Reply:
this is good!